docs/understanding-dependencies.md

```diff
@@ -0,0 +1,24 @@
+# dependencies vs. devDependencies
+
+> What's the difference between these two properties in `package.json`?
+
+While functionally almost identical, there are small differences that set these two apart. At its core, both `dependencies` and `devDependences` properties in `package.json` are where external packages that your project uses are defined. Both properties contain an object of name/versions. Each time you `npm install` or `npm install --save-dev` a new entry is added to either `dependencies` or `devDependencies` respectively, in the format of 
+```json
+  "name": "version"
+```
+
+## dependencies
+
+The `dependencies` property is where external packages that are essential for your code to run in production are defined (e.g React and React-dom).
+
+## devDependencies
+
+In contrast, `devDependencies` property is *usually* (but not always) used to define packages needed for development. Packages that could be considered a `devDependency` include tooling for Unit Tests (Jest, Mocha, Chai, Ava), linting (Eslint, Prettier), Javascript transpilation (Babel, TypeScript, Flow), minification (UglifyJs), and deployment tools.
```

Comment:
I have a couple notes about this paragraph:

1. I'm guessing most beginners won't know what all of these dependencies are, might create more questions than answers.

2. I saw that the caveats explain below that the difference is semantic, but I think it'd be good to have it within the same section; I feel like caveats are easy to miss, folks might stop reading once they get through this paragraph.

What do you think about this?

> In contrast, `devDependencies` are generally packages needed for development, but not used in your main codebase. 
>
> For example, many developers find it worthwhile to write automated tests. Tests let the developer run a task to quickly verify that core functionality still works, without having to manually try things out in the browser.
>
> For React applications, a common test libary is [jest](https://jestjs.io/). Jest is an external dependency, but it's not one we want to include in our application code; our users don't need to run the tests, it's only necessary during development. So we can add it to `devDependencies`, as a way to indicate to other developers that this dependency is only useful for development.
>
> **Note:** For modern web applications, like the ones created with Guppy, the difference between `dependencies` and `devDependencies` is semantic. There is no functional difference between the two. This is because applications only bundle the dependencies _actively used_ in the project. Simply installing a dependency, whether to `dependencies` or `devDependencies`, doesn't affect the code sent to your users, unless you use that dependency in the application.

Reply:
Yeah, re-reading what I wrote - I definitely agree with you RE: beginners not knowing everything listed. I think the above flows much better and is way more beginner friendly.

Reply:
It's a serious challenge to stay aware of how much accrued knowledge we have, as experienced devs!

In the first version I shared on Twitter of Guppy, the placeholder text for "Project Name" was "Todo MVC". My (not-a-programmer) boyfriend looked at it and said "What's Todo MVC?", and I realized that my placeholder text would be complete nonsense to most newcomers, haha.